### PR TITLE
cargo update

### PR DESCRIPTION
**Pull request by curiousleo**
Update deps (`cargo update; ./nix/update-carnix.sh`) and get rid of one deprecation warning (`Vec1::from_vec` -> `Vec1::try_from_vec`).

### CODE REVIEW

**Comment by curiousleo** [comment]
PTAL

**Comment by Profpatsch** [comment]
Omnom, that sweet 2k carnix diff. Why do we need to update the dependencies?

File: Cargo.lock [hunk 1]
```diff
@@ -316,40 +329,40 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 
 [[package]]
 name = "libc"
-version = "0.2.55"
+version = "0.2.65"
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/rust-lang/libc/compare/0.2.55...0.2.65

Various improvements, it's a very active project!

File: Cargo.lock [hunk 2]
```diff
@@ -316,40 +329,40 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
 source = "registry+https://github.com/rust-lang/crates.io-index"
 
 [[package]]
 name = "log"
-version = "0.4.6"
+version = "0.4.8"
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/rust-lang/log/blob/master/CHANGELOG.md

Better error messages, docs use 2018 edition syntax.

File: Cargo.lock [hunk 3]
```diff
@@ -316,40 +329,40 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
  "notify 5.0.0-pre.1 (registry+https://github.com/rust-lang/crates.io-index)",
- "proptest 0.9.1 (registry+https://github.com/rust-lang/crates.io-index)",
- "regex 1.1.0 (registry+https://github.com/rust-lang/crates.io-index)",
- "serde 1.0.88 (registry+https://github.com/rust-lang/crates.io-index)",
- "serde_derive 1.0.88 (registry+https://github.com/rust-lang/crates.io-index)",
- "serde_json 1.0.38 (registry+https://github.com/rust-lang/crates.io-index)",
- "structopt 0.2.14 (registry+https://github.com/rust-lang/crates.io-index)",
- "tempfile 3.0.7 (registry+https://github.com/rust-lang/crates.io-index)",
- "vec1 1.1.0 (registry+https://github.com/rust-lang/crates.io-index)",
+ "proptest 0.9.4 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/AltSysrq/proptest/blob/master/proptest/CHANGELOG.md

Performance improvements.

File: Cargo.lock [hunk 4]
```diff
@@ -316,40 +329,40 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
+ "regex 1.3.1 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/rust-lang/regex/blob/master/CHANGELOG.md

So many bugfixes!

File: Cargo.lock [hunk 5]
```diff
@@ -316,40 +329,40 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
+ "serde 1.0.103 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/serde-rs/serde/compare/v1.0.88...v1.0.103

Removes some clippy warnings, adds serialization for atomic integer types, improved test coverage.

File: Cargo.lock [hunk 6]
```diff
@@ -316,40 +329,40 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
+ "serde_derive 1.0.103 (registry+https://github.com/rust-lang/crates.io-index)",
+ "serde_json 1.0.42 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/serde-rs/json/compare/v1.0.38...v1.0.42

Bugfixes.

File: Cargo.lock [hunk 7]
```diff
@@ -316,40 +329,40 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
+ "structopt 0.2.18 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/TeXitoi/structopt/blob/master/CHANGELOG.md

Bugfix.

File: Cargo.lock [hunk 8]
```diff
@@ -316,40 +329,40 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
+ "tempfile 3.1.0 (registry+https://github.com/rust-lang/crates.io-index)",
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/Stebalien/tempfile/compare/v3.0.7...v3.1.0

Bugfixes.

File: Cargo.lock [hunk 9]
```diff
@@ -364,21 +377,20 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
 
 [[package]]
 name = "memchr"
-version = "2.2.0"
+version = "2.2.1"
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/BurntSushi/rust-memchr/compare/2.2.0...2.2.1

Performance improvements.

File: Cargo.lock [hunk 10]
```diff
@@ -364,21 +377,20 @@ source = "registry+https://github.com/rust-lang/crates.io-index"
...
 source = "registry+https://github.com/rust-lang/crates.io-index"
 
 [[package]]
 name = "mio"
-version = "0.6.16"
+version = "0.6.19"
```
Comment (curiousleo) [inline Cargo.lock]: https://github.com/tokio-rs/mio/blob/master/CHANGELOG.md

Bugfixes.

**Comment by Profpatsch** [comment]
Okayokay, thanks :)
Great move checking the deps, I think we should check at least the changelogs of major version bumps when we update. iirc cargo doesn’t let you do a minor version bump if the API changes anyway.

**Comment by curiousleo** [comment]
> I think we should check at least the changelogs of major version bumps when we update.

Totally agree.

**Pull request merged**